Replaced unnecessary setUp() in the test suite.

charettes opened the pull request:
Whether or not it was safe to switch to `setUpTestData()` was investigated locally when working on a [per-test isolation feature of `setUpTestData()`](https://github.com/charettes/django-testdata) I'm planning to discuss on django-developers in a few hours.

felixxm commented on tests/admin_views/tests.py:
```diff
@@ -3687,21 +3687,17 @@ class AdminInlineFileUploadTest(TestCase):
+        cls.gallery = Gallery(name="Test Gallery")
+        cls.gallery.save()
+        cls.picture = Picture(name="Test Picture", image=filename, gallery=cls.gallery)
+        cls.picture.save()
```
Maybe `.objects.create(...)`? We will have two lines less :smile: Please change to single quotes.

felixxm commented on tests/admin_views/tests.py:
```diff
@@ -3740,6 +3736,8 @@ class AdminInlineTests(TestCase):
     @classmethod
     def setUpTestData(cls):
         cls.superuser = User.objects.create_superuser(username='super', password='secret', email='super@example.com')
+        cls.collector = Collector(pk=1, name='John Fowles')
```
`.objects.create(...)`

felixxm commented on tests/foreign_object/tests.py:
```diff
@@ -18,28 +18,28 @@
-        self.george = Person.objects.create(name='George', person_country=self.usa)
+        cls.bob = Person()
+        cls.bob.name = 'Bob'
+        cls.bob.person_country = cls.usa
```
We can move `name` and `person_country` to `Person(...)`.

felixxm commented on tests/postgres_tests/test_array.py:
```diff
@@ -366,16 +367,17 @@ def test_unsupported_lookup(self):
+        cls.dates = [now.date()]
+        cls.times = [now.time()]
+        cls.objs = [
             DateTimeArrayModel.objects.create(
```
This can be single-lined.

felixxm commented on tests/sites_tests/tests.py:
```diff
@@ -20,13 +20,14 @@
-        self.site = Site(
+    @classmethod
+    def setUpTestData(cls):
+        cls.site = Site(
```
This can be single-lined.

charettes commented:
@felixxm I don't mind doing all of these changes in following commits but I wanted to keep the noise ratio as small as possible when doing the `setUp(self)` to `setUpTestData(cls)` switch.

charettes commented:
@felixxm I just pushed 1c9af5ed1ceb60ab3ea0117ede9e1ae898dc8e5a to deal with your suggestions. I'll let fellows decide whether or not they want to squash it in 2d1d49d95ee08ea5f81e55755e6412971e548610.

felixxm commented:
Thanks Simon :+1:

charettes commented on tests/auth_tests/test_decorators.py:
```diff
@@ -58,12 +58,15 @@ class PermissionsRequiredDecoratorTest(TestCase):
+        cls.user.user_permissions.add(*perms)
+
+    def setUp(self):
+        self.factory = RequestFactory()
```
No it wasn't. I'll keep it in `setUp` in all instances.

charettes commented:
@timgraham, I addressed your comments and pushed a following commit that defines reused `RequestFactory` instances as class attributes. That removes a few other unnecessary `setUp`.
